addons/addon-base-ui/packages/base-ui/src/parts/ForceLogout.js

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+import { observable, action, decorate, computed, runInAction } from 'mobx';
+import { observer, inject } from 'mobx-react';
+import { Button, Modal } from 'semantic-ui-react';
+
+const jwtDecode = require('jwt-decode');
+
+// expected props
+// - authentication
+// - app
+class ForceLogout extends React.Component {
+  constructor(props) {
+    super(props);
+    runInAction(() => {
+      this.tokenActive = true;
+    });
+  }
+
+  get app() {
+    return this.props.app;
+  }
+
+  get authentication() {
+    return this.props.authentication;
+  }
+
+  get modalOpen() {
+    return !this.tokenActive;
+  }
+
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      const expirationInMins = this.hasTokenExpired();
+      // eslint-disable-next-line react/no-unused-state
+      if (expirationInMins > 0) {
+        runInAction(() => {
+          this.tokenActive = true;
+        });
+      } else {
+        runInAction(() => {
+          this.tokenActive = false;
+        });
+      }
+    }, 1000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+  }
+
+  doLogout = async () => {
+    clearInterval(this.timer);
+    return this.authentication.logout({ autoLogout: true });
+  };
+
+  hasTokenExpired = () => {
+    let minutesRemaining;
+    try {
+      const idToken = localStorage.getItem('appIdToken');
+      const decodedIdToken = jwtDecode(idToken);
+      const expiresAt = _.get(decodedIdToken, 'exp', 0) * 1000;
```

Comment:
this can just be `return _.get(decodedIdToken, 'exp', Date.now()) >= Date.now())`

Reply:
Wanted to still keep this inside the try-catch block in case the token was not received at all, or if it was corrupted.

Reply:
I was thinking something like this. Thoughts?
```
hasTokenExpired = () => {
    try {
      const idToken = localStorage.getItem('appIdToken');
      const decodedIdToken = jwtDecode(idToken);
      return _.get(decodedIdToken, 'exp', Date.now()) >= Date.now())
    } catch (e) {
      return false;
    }
  };
  ```

Reply:
We can do it this way too. I originally wanted to convey to the reader computation is being done by the number of minutes left until token expiration.